src/block/block_read.c

```diff
@@ -215,9 +215,12 @@ __wt_block_read_off(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_ITEM *buf, uin
         if (!F_ISSET(session, WT_SESSION_QUIET_CORRUPT_FILE))
             __wt_errx(session,
               "%s: potential hardware corruption, read checksum error for %" PRIu32
-              "B block at offset %" PRIuMAX
-              ": calculated block checksum doesn't match expected checksum",
-              block->name, size, (uintmax_t)offset);
+              "B block at offset %" PRIuMAX ": calculated block checksum of %#" PRIx32
+              " doesn't match expected checksum of %#" PRIx32,
+              block->name, size, (uintmax_t)offset,
+              __wt_checksum(
+                buf->mem, F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP),
```

Comment:
Can you put `F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP)` into a variable, perhaps `size_t check_size;` ?

Reply:
Yeah, that would be neater, thanks.